tests/e2e/macvtap_test.go

```diff
@@ -231,6 +238,22 @@ func filterPods(pods []v1.Pod, filterFunction func(v1.Pod) bool) []v1.Pod {
 	return filteredPods
 }
 
+func findMacvtapNamespace() (string, error) {
+	macvtapDaemonSetName := "macvtap-cni"
+	daemonSetList, err := clientset.ExtensionsV1beta1().DaemonSets(v1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
```

Comment:
I don't follow; you list the daemonsets, then iterate them looking for a specific one. Why don't you just `Get` the correct daemonset (instead of `List`ing then iterating) ?

Reply:
yeah but when I did that I got an error that I can't `Get` with a specific name and v1.NamespaceAll..:
```
error: {
                  cause: {
                      s: "an empty namespace may not be set when a resource name is provided",
                  },
                  msg: "Failed to get daemonSet in all namespaces",
              },

```
this was the next best thing.. but I agree it's odd. Do you have an alternative solution?

Reply:
Right! Sorry for putting you on the wrong track.

I think using a field selector would then be the correct thing; does the example below work ?

```golang
daemonSetList, err := clientset.ExtensionsV1beta1().DaemonSets(v1.NamespaceAll).List(context.TODO(), metav1.ListOptions{
	FieldSelector:        fmt.Sprintf("metadata.name=%s", macvtapDaemonSetName),
})
```

Reply:
totally agree :)